lib/src/Pos/Node/API.hs

```diff
@@ -547,47 +553,209 @@ instance ToSchema (V1 Version) where
         pure $ NamedSchema (Just "V1Version") $ mempty
             & type_ .~ SwaggerString
 
+
+newtype SecurityParameter = SecurityParameter Int
+    deriving (Show, Eq, Generic, ToJSON, FromJSON, Arbitrary)
+
+instance Buildable SecurityParameter where
+    build (SecurityParameter i) = bprint shown i
+
+instance ToSchema SecurityParameter where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "SecurityParameter") $ mempty
+            & type_ .~ SwaggerNumber
+            & maximum_ .~ Just (fromIntegral (maxBound :: Int))
+            & minimum_ .~ Just (fromIntegral (minBound :: Int))
+
+
+instance ToSchema (V1 Core.SlotId) where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Core.SlotId") $ mempty
+            & type_ .~ SwaggerObject
+            & required .~ ["slot", "epoch"]
+            & properties .~ (mempty
+                & at "slot" ?~ (Inline $ mempty
+                    & type_ .~ SwaggerNumber
+                    & maximum_ .~ Just (fromIntegral (maxBound :: Word16))
+                    & minimum_ .~ Just (fromIntegral (minBound :: Word16))
+                    )
+                & at "epoch" ?~ (Inline $ mempty
+                    & type_ .~ SwaggerNumber
+                    & maximum_ .~ Just (fromIntegral (maxBound :: Word64))
+                    & minimum_ .~ Just (fromIntegral (minBound :: Word64))
+                    )
+                )
+
+instance ToJSON (V1 Core.SlotId) where
+    toJSON (V1 s) =
+        object
+            [ "epoch" .= toJSON (Core.getEpochIndex $ Core.siEpoch s)
+            , "slot"  .= toJSON (Core.getSlotIndex  $ Core.siSlot s)
+            ]
```

Comment:
Can't we rely on `Core.SlotId` already existing instance for this? 

Actually.. that's a very good point. Why are we using `V1` wrapper here :thinking: ? This wrapper exists to avoid orphan instances in the wallet code, but here, we are onto the node so we could just define those instances in the right places, next to the corresponding types!

Reply:
> next to the corresponding types!

Are you saying 1) that we can modify existing JSON instances in Core, 2) that we should do `newtype SlotId = ...` here, or something else?

JSON encoded `Core.SlotId`:
```json
{"siEpoch":4419942607164,"siSlot":{"getSlotIndex":32}}
```

JSON encoded `V1 Core.SlotId`:
```json
"{\"slot\":32,\"epoch\":4419942607164}"
```